server/src/services/library.service.ts

```diff
@@ -249,12 +249,17 @@ export class LibraryService {
         if (!dto.name) {
           dto.name = 'New External Library';
         }
+        dto.isReadOnly ??= true;
         break;
       }
       case LibraryType.UPLOAD: {
         if (!dto.name) {
           dto.name = 'New Upload Library';
         }
+        dto.isReadOnly ??= false;
+        if (dto.isReadOnly) {
+          throw new BadRequestException('Upload libraries cannot be readonly');
+        }
```

Comment:
This dto field doesn't seem to get used - I think you mean to use it at the end.

Also just a preference thing but I think `isReadOnly: dto.isReadOnly ?? dto.type !== LibraryType.UPLOAD` at the end reads better than `dto.isReadOnly ??= true/false`.

Reply:
thanks!! updated